src/components/Base/Tooltip/index.jsx

```diff
@@ -0,0 +1,164 @@
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      visible: props.visible,
+    };
+    this.disabled = props.disabled;
+
+    const placementMapper = {
+      left: 'left',
+      right: 'right',
+      top: 'top',
+      bottom: 'bottom',
+    };
+    this.placement = placementMapper[this.props.placement] || 'bottom';
+
+    const triggerMapper = {
+      focus: 'focus',
+      click: 'click',
+      hover: 'hover',
+    };
+    this.trigger = triggerMapper[this.props.trigger] || 'hover';
+  }
+
+  showPopper = () => {
+    if (this.disabled) return;
+    this.setState({ visible: true });
+  }
+
+  hidePopper = () => {
+    this.setState({ visible: false });
+  }
+
+  handleTogglePopper = () => {
+    this.state.isShow ? this.hidePopper() : this.showPopper();
+  }
+
+  handleDocumentClick = (e) => {
+    const rootNode = findDOMNode(this);
+
+    if (!rootNode || rootNode.contains(e.target)
+        || !this.target || this.target.contains(e.target)
+        || !this.popper || this.popper.contains(e.target)) {
+      return;
+    }
+    this.hidePopper();
+  }
+
+  bindEvent = () => {
+    const target = this.target;
+    const popper = this.popper;
+
+    if (!target || !popper) return;
+
+    if (this.trigger === 'hover') {
+      target.addEventListener('mouseenter', this.showPopper);
+      target.addEventListener('mouseleave', this.hidePopper);
+      popper.addEventListener('mouseenter', this.showPopper);
+      popper.addEventListener('mouseleave', this.hidePopper);
+    } else {
+      target.addEventListener('click', this.handleTogglePopper);
+      document.addEventListener('click', this.handleDocumentClick);
+    }
+  }
+
+  removeEvent = () => {
+    const target = this.target;
+    const trigger = this.trigger;
+
+    if (!this.target) return;
+    if (trigger === 'focus') {
+      target.removeEventListener('focus', this.showPopper);
+      target.removeEventListener('blur', this.hidePopper);
+    } else if (trigger === 'click') {
+      target.removeEventListener('click', this.handleTogglePopper);
+      document.removeEventListener('click', this.handleDocumentClick);
+    } else {
+      target.removeEventListener('mouseenter', this.showPopper);
+      target.removeEventListener('mouseleave', this.hidePopper);
+    }
+  }
+
+  initPopover = () => {
+    if (this.disabled) return;
+
+    this.bindEvent();
+  }
+
+  destroy = () => {
+    this.removeEvent();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ('visible' in nextProps) {
+      this.setState({ visible: nextProps.visible });
+    }
+  }
+
+  componentDidMount() {
+    this.initPopover();
+  }
+
+  componentWillUnmount() {
+    this.destroy();
+  }
+
+  render() {
+    const { prefixCls, className, content, children } = this.props;
+
+    return (
+      <Manager
+        className={classNames(styles.tooltip, {
+          [className]: className,
+        })}
+      >
+        <Target
+          innerRef={c => { this.target = findDOMNode(c); }}
```

Comment:
c is the DOM node, no need to call findDOMNode ?

Reply:
yes